sklearn/manifold/_t_sne.py

```diff
@@ -714,16 +733,28 @@ def _fit(self, X, skip_num_points=0):
                 if self.verbose:
                     print("[t-SNE] Computing pairwise distances...")
 
-                if self.metric == "euclidean":
-                    distances = pairwise_distances(X, metric=self.metric,
-                                                   squared=True)
+                if self.metric == 'euclidean':
+                    if self.square_distance in [True, 'legacy']:
+                        # Euclidean is squared here, rather than using **= 2,
+                        # because euclidean_distances already calculates
+                        # squared distances, and returns np.sqrt(dist) for
+                        # squared=False. So, using dist **= 2 would result in:
+                        #   ^2 -> sqrt() -> ^2
+                        distances = pairwise_distances(X, metric=self.metric,
+                                                       squared=True)
+                    else:
+                        distances = pairwise_distances(X, metric=self.metric)
                 else:
+                    # Euclidean has separate calls as it's slower for n_jobs>1
                     distances = pairwise_distances(X, metric=self.metric,
                                                    n_jobs=self.n_jobs)
 
-                if np.any(distances < 0):
-                    raise ValueError("All distances should be positive, the "
-                                     "metric given is not correct")
+            if self.metric != 'euclidean' and self.square_distance is True:
```

Comment:
I believe this change would cause non-Euclidean distances to be squared when `square_distance=='legacy'`, because the truth value for non-empty strings is considered True.

Reply:
right I forgot it was not just a boolean

Reply:
No worries! :)